This proposed upstream patch addresses

	18562372 Failed to create a new project under Horizon

and is tracked under Launchpad bug 1264228. Although it's been
addressed in Icehouse 2014.1, the patch below is still not yet released
for Havana.  It has been modified to apply cleanly into our current
Havana implementation

From e02eaab30996af7e8770cd651bd8aa7d504358a8 Mon Sep 17 00:00:00 2001
From: JiaHao Li <kaho_lai@hotmail.com>
Date: Thu, 26 Dec 2013 15:37:14 +0800
Subject: [PATCH] Sync OPENSTACK_KEYSTONE_DEFAULT_ROLE with keystone

For now, keystone default role is _member_, while horizon set
OPENSTACK_KEYSTONE_DEFAULT_ROLE to Member. It will really be user
friendly to modify horizon default value to _member_ to sync with
keystone's default setting.

Conflicts:

    doc/source/topics/settings.rst

Change-Id: I55d15e6cfb74e52e933c5a44efd6c27930415738
Closes-Bug: #1264228
(cherry picked from commit 0aacc44f324c3db049f912da1f84d93c1142cb37)
---
 doc/source/topics/settings.rst                     |    2 +-
 .../local/local_settings.py.example                |    2 +-
 openstack_dashboard/test/settings.py               |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/doc/source/topics/settings.rst b/doc/source/topics/settings.rst
index 1f5eeea..2b2eabd 100644
--- a/doc/source/topics/settings.rst
+++ b/doc/source/topics/settings.rst
@@ -176,7 +176,7 @@ If you do not have multiple regions you should use the ``OPENSTACK_HOST`` and
 ``OPENSTACK_KEYSTONE_DEFAULT_ROLE``
 -----------------------------------
 
-Default: "Member"
+Default: "_member_"
 
 The name of the role which will be assigned to a user when added to a project.
 This name must correspond to a role name in Keystone.
diff --git a/openstack_dashboard/local/local_settings.py.example b/openstack_dashboard/local/local_settings.py.example
index a1a8474..92fcc16 100644
--- a/openstack_dashboard/local/local_settings.py.example
+++ b/openstack_dashboard/local/local_settings.py.example
@@ -126,7 +126,7 @@ EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
 
 OPENSTACK_HOST = "127.0.0.1"
 OPENSTACK_KEYSTONE_URL = "http://%s:5000/v2.0" % OPENSTACK_HOST
-OPENSTACK_KEYSTONE_DEFAULT_ROLE = "Member"
+OPENSTACK_KEYSTONE_DEFAULT_ROLE = "_member_"
 
 # Disable SSL certificate checks (useful for self-signed certificates):
 # OPENSTACK_SSL_NO_VERIFY = True
diff --git a/openstack_dashboard/test/settings.py b/openstack_dashboard/test/settings.py
index 85f470d..08086a2 100644
--- horizon-2013.2.3/openstack_dashboard/test/settings.py.~1~	2014-04-03 11:45:53.000000000 -0700
+++ horizon-2013.2.3/openstack_dashboard/test/settings.py	2014-05-19 11:50:14.914650963 -0700
@@ -85,7 +85,7 @@
 }
 
 OPENSTACK_KEYSTONE_URL = "http://localhost:5000/v2.0"
-OPENSTACK_KEYSTONE_DEFAULT_ROLE = "Member"
+OPENSTACK_KEYSTONE_DEFAULT_ROLE = "_member_"
 
 OPENSTACK_KEYSTONE_MULTIDOMAIN_SUPPORT = True
 OPENSTACK_KEYSTONE_DEFAULT_DOMAIN = 'test_domain'
